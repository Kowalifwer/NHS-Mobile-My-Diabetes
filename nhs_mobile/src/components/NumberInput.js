import React from 'react';
import GlobalStyle from '../styles/GlobalStyle';
import { TextInput } from 'react-native';


class NumberInput extends React.Component {
    constructor(props, diary_name) {
        super();
        this.state = { input_text: "", last_input_text: "" }
        this.diary_name = diary_name
    }

    validateInput(input) {
        console.log("testing: ", input)
        if (input.match(/(^$)|(^\d+\.*\d*?$)/) == null) {
            return false
        } else {
            return true
        }
    }

    updateInput(input) {
        if (this.validateInput(input)) {
            this.setState({last_input_text: this.state.input_text, input_text: input})
        }
    }

    render() {
        return (
            <TextInput
                style={ GlobalStyle.InputField }
                value={ this.state.input_text }
                keyboardType = 'numeric'
                placeholder={ this.props.label }
                onChangeText={ value => {
                    if (this.validateInput(value)) {
                        this.updateInput(value)
                    }
                }}
            />
        );
    }
}

export default NumberInput;