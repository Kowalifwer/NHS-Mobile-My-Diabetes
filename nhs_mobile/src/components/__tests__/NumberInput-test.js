import React from 'react';
import renderer from 'react-test-renderer';
import NumberInput from '../NumberInput';

it('renders correctly across screens', () => {
    const tree = renderer.create(<NumberInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });