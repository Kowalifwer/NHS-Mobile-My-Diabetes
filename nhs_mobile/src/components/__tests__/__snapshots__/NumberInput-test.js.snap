// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="numeric"
  onChangeText={[Function]}
  rejectResponderTermination={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#555",
      "borderRadius": 10,
      "borderWidth": 1,
      "fontSize": 20,
      "height": 40,
      "marginBottom": 10,
      "marginTop": 10,
      "paddingBottom": 5,
      "textAlign": "center",
      "width": 350,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
